src/content/posts/whats-possible-with-melange.mdx

```diff
@@ -0,0 +1,202 @@
+## Different configs for development and production
+
+Seems basic to showcase this, but previously in BuckleScript you can’t setup different config for production and development. That’s not the case for dune, where you are able to have different [profiles]() and set everything you need differently. Warnings, instrumentation and some libraries are the most common to load differently.
+
+Previously we had a script that generates the config by certain parameter and accomplish such a basic functionality.
+
+## Integration with documentation tooling (odoc)
+
+The other benefit of integrating with the OCaml platform is the documentation tooling is available. It’s called [**odoc**](https://github.com/ocaml/odoc) and it will generate a documentation website from your interface files, with navigation, cross-linking and many necessities
+
+The other cool thing is that eventually can be published under ocaml.org, for example [ocaml.org/p/melange-json](https://ocaml.org/p/melange-json/latest).
+
+## Downsides
+
+Those are massive improvements and gives the possibility to create a new way of working with Reason, currently at Ahrefs we have been exploring many ways of pushing this environment forward. With that being said, there are a few downsides from the lenses of ReScript.
+
+## Learning a new tool
+
+The dune integration brings power to users, but with a cost of learning a new tool, **dune**. Even being powerful, it’s also has a few rough edges, and the learning about those might be noticeable.
+
+The initial blocker might be their syntax, which is very lisp-y. Hopefully your editor can support colorized parentesis to overcome it.
+
+```clojure
+(library
+ (name url)
+ (wrapped false)
+ (libraries uri js))
+```
+
+It’s mainained by [Jane Street](https://www.janestreet.com/), [Tarides](https://tarides.com/), [OCamlPro](https://ocamlpro.com) where they improve it frequently and they are [open for feedback and feature requests](https://discuss.ocaml.org/t/dune-wish-list-for-2023/11083).
+
+After working with it almost 10 months I’m convinced that it’s a big improvement over previous status.
+
+To learn more about dune specifically, I recommend → [A handy guide to Dune](https://mukulrathi.com/ocaml-tooling-dune/)
+
+## Different package manager: opam
+
+The recommended package manager for Melange development is **opam**. **opam** is OCaml's package manager (think of npm, rubygems, pip, cargo) and compiler version manager (nvm, rvm, …). In opam, those compiler version plus the dependencies are called switches.
+
+Again, a new tool. It has a few key differences from **npm** based on a safer model:
+
+- **Safer**. The premise of opam is to ensure that after a successful installation, the project will compile correctly.
+- **More strict**. You only can have on single version of a dependency on your project
+- **but slower**. Since it downloads and builds all dependencies from source, the installation takes much longer than npm since does a bunch of more work.
```

Comment:
```suggestion
- **but slower**. Since it downloads and builds all dependencies from source, the installation takes much longer than npm since it does a bunch more work.
```